src/plugins/cache.js

```diff
@@ -107,6 +126,14 @@ export default class Cache {
 
   async fetchManagers (...managers: Manager[]): Promise<Plugin[]> {
     let plugins = []
+    if (this.cache.node_version !== process.version) {
+      for (let manager of managers) {
+        await manager.handleNodeVersionChange()
```

Comment:
@dickeyxxx wrapped in the lock in https://github.com/heroku/cli-engine/pull/188/commits/acbf019f8e5637ba9ee58da0c0b16c158c8c8bbf